cmd/juju/secrets/list.go

```diff
@@ -0,0 +1,177 @@
+	c.out.AddFlags(f, "tabular", map[string]cmd.Formatter{
+		"yaml": cmd.FormatYaml,
+		"json": cmd.FormatJson,
+		"tabular": func(writer io.Writer, value interface{}) error {
+			return formatSecretsTabular(writer, value)
+		},
+	})
+}
+
+type secretValueDetails struct {
+	Data  secrets.SecretData `json:",omitempty,inline" yaml:",omitempty,inline"`
+	Error error              `json:"error,omitempty" yaml:"error,omitempty"`
+}
+
+type secretDisplayDetails struct {
+	ID          int                 `json:"ID" yaml:"ID"`
+	Revision    int                 `json:"revision" yaml:"revision"`
+	Path        string              `json:"path" yaml:"path"`
+	Scope       secrets.Scope       `json:"scope" yaml:"scope"`
+	Version     int                 `json:"version" yaml:"version"`
+	Description string              `json:"description,omitempty" yaml:"description,omitempty"`
+	Tags        map[string]string   `json:"tags,omitempty" yaml:"tags,omitempty"`
+	ProviderID  string              `json:"provider-id,omitempty" yaml:"provider-id,omitempty"`
+	CreateTime  time.Time           `json:"create-time" yaml:"create-time"`
+	UpdateTime  time.Time           `json:"update-time" yaml:"update-time"`
+	Value       *secretValueDetails `json:"value,omitempty" yaml:"value,omitempty"`
+}
+
+// Run implements cmd.Run.
+func (c *listSecretsCommand) Run(ctxt *cmd.Context) error {
+	if c.showSecrets && c.out.Name() == "tabular" {
+		ctxt.Infof("secret values are not shown in tabular format")
+		c.showSecrets = false
+	}
+
+	api, err := c.listSecretsAPIFunc()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer api.Close()
+
+	secrets, err := api.ListSecrets(c.showSecrets)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	details := make([]secretDisplayDetails, len(secrets))
+	for i, m := range secrets {
+		details[i] = secretDisplayDetails{
+			Path:        m.Metadata.Path,
+			Scope:       m.Metadata.Scope,
+			Version:     m.Metadata.Version,
+			Description: m.Metadata.Description,
+			Tags:        m.Metadata.Tags,
+			ID:          m.Metadata.ID,
+			ProviderID:  m.Metadata.ProviderID,
+			Revision:    m.Metadata.Revision,
+			CreateTime:  m.Metadata.CreateTime,
+			UpdateTime:  m.Metadata.UpdateTime,
+		}
+		if c.showSecrets && m.Value != nil {
+			details[i].Value = &secretValueDetails{
+				Error: m.Error,
+			}
+			if m.Error != nil {
+				continue
+			}
+			val, err := m.Value.Values()
+			if err != nil {
+				details[i].Value.Error = err
+			} else {
+				details[i].Value.Data = val
+			}
+		}
+	}
+	return c.out.Write(ctxt, details)
+}
+
+// formatSecretsTabular writes a tabular summary of secret information.
+func formatSecretsTabular(writer io.Writer, value interface{}) error {
+	secrets, ok := value.([]secretDisplayDetails)
+	if !ok {
+		return errors.Errorf("expected value of type %T, got %T", secrets, value)
+	}
+
+	tw := output.TabWriter(writer)
+	w := output.Wrapper{tw}
+	w.SetColumnAlignRight(1)
+
+	w.Println("ID", "Scope", "Revision", "Path")
```

Comment:
Does path show which backend is used? Tabular should probably print out that this model's secrets are stored in the controller vs HC vault

Reply:
Also tabular should show an age field/last changed/last accessed. Important for auditing.

e.g. 
ID | Scope | Revision | Path | Age
1 bla 12 secret://... 5d